gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GCSManualClientImpl.java

```diff
@@ -0,0 +1,222 @@
+      GoogleCloudStorageItemInfo itemInfo, GoogleCloudStorageReadOptions readOptions)
+      throws IOException {
+    return gcsClientDelegate.open(itemInfo, readOptions);
+  }
+
+  @Override
+  public void deleteBuckets(List<String> bucketNames) throws IOException {
+    gcsClientDelegate.deleteBuckets(bucketNames);
+  }
+
+  @Override
+  public void deleteObjects(List<StorageResourceId> fullObjectNames) throws IOException {
+    gcsClientDelegate.deleteObjects(fullObjectNames);
+  }
+
+  @Override
+  public void copy(
+      String srcBucketName,
+      List<String> srcObjectNames,
+      String dstBucketName,
+      List<String> dstObjectNames)
+      throws IOException {
+    gcsClientDelegate.copy(srcBucketName, srcObjectNames, dstBucketName, dstObjectNames);
+  }
+
+  @Override
+  public List<String> listBucketNames() throws IOException {
+    return gcsClientDelegate.listBucketNames();
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> listBucketInfo() throws IOException {
+    return gcsClientDelegate.listBucketInfo();
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> listObjectInfo(
+      String bucketName, String objectNamePrefix, ListObjectOptions listOptions)
+      throws IOException {
+    return gcsClientDelegate.listObjectInfo(bucketName, objectNamePrefix, listOptions);
+  }
+
+  @Override
+  public ListPage<GoogleCloudStorageItemInfo> listObjectInfoPage(
+      String bucketName, String objectNamePrefix, ListObjectOptions listOptions, String pageToken)
+      throws IOException {
+    return gcsClientDelegate.listObjectInfoPage(
+        bucketName, objectNamePrefix, listOptions, pageToken);
+  }
+
+  @Override
+  public GoogleCloudStorageItemInfo getItemInfo(StorageResourceId resourceId) throws IOException {
+    return gcsClientDelegate.getItemInfo(resourceId);
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> getItemInfos(List<StorageResourceId> resourceIds)
+      throws IOException {
+    return gcsClientDelegate.getItemInfos(resourceIds);
+  }
+
+  @Override
+  public List<GoogleCloudStorageItemInfo> updateItems(List<UpdatableItemInfo> itemInfoList)
+      throws IOException {
+    return gcsClientDelegate.updateItems(itemInfoList);
+  }
+
+  @Override
+  public void compose(
+      String bucketName, List<String> sources, String destination, String contentType)
+      throws IOException {
+    gcsClientDelegate.compose(bucketName, sources, destination, contentType);
+  }
+
+  @Override
+  public GoogleCloudStorageItemInfo composeObjects(
+      List<StorageResourceId> sources, StorageResourceId destination, CreateObjectOptions options)
+      throws IOException {
+    return gcsClientDelegate.composeObjects(sources, destination, options);
+  }
+
+  @Override
+  public Map<String, Long> getStatistics() {
+    return gcsClientDelegate.getStatistics();
+  }
+
+  @Override
+  public void close() {
+    gcsClientDelegate.close();
+  }
+
+  public static class GCSManualClientImplBuilder {
```

Comment:
It might be a good idea to move this to GoogleCloudStorage class so that we can hide this Impl class details.